Qt Bugreports PYSIDE-1272
TypeError: must be str, not getset_descriptor
---
diff --git a/sources/shiboken2/shibokenmodule/files.dir/shibokensupport/signature/errorhandler.py b/sources/shiboken2/shibokenmodule/files.dir/shibokensupport/signature/errorhandler.py
index cb14883....0403358 100644

--- a/shibokenmodule/files.dir/shibokensupport/signature/errorhandler.py
+++ b/shibokenmodule/files.dir/shibokensupport/signature/errorhandler.py
@@ -56,6 +56,8 @@
 This matter will be improved in a later version.
 """
 
+import sys
+
 from shibokensupport.signature import inspect
 from shibokensupport.signature import get_signature
 from shibokensupport.signature.mapping import update_mapping, namespace
@@ -122,6 +122,11 @@
     # We don't raise the error here, to avoid the loader in the traceback.
     return TypeError, msg
 
+def check_string_type(s):
+    if sys.version_info[0] == 3:
+        return isinstance(s, str)
+    else:
+        return isinstance(s, (str, unicode))
 
 def make_helptext(func):
     existing_doc = func.__doc__
@@ -135,7 +140,7 @@
     except AttribureError:
         func_name = func.__func__.__name__
     sigtext = "\n".join(func_name + str(sig) for sig in sigs)
-    msg = sigtext + "\n\n" + existing_doc if existing_doc else sigtext
+    msg = sigtext + "\n\n" + existing_doc if check_string_type(existing_doc) else sigtext
     return msg
 
 # end of file
